test/performance/io/format_fastq_benchmark.cpp

```diff
@@ -0,0 +1,242 @@
+    auto qual = test::generate_sequence<phred42>(default_sequence_length, 0, seed);
+
+    for (auto _ : state)
+    {
+        format.write(ostream, options, seq, id, qual);
+    }
+}
+
+// ============================================================================
+// read dummy fastq file from a stream
+// ============================================================================
+void fastq_read_from_stream(benchmark::State & state)
+{
+    sequence_file_format_fastq format;
+    sequence_file_input_options<dna5, false> const options{};
+
+    std::string id{};
+    std::vector<dna5> seq{};
+    std::vector<phred42> qual{};
+
+    size_t n_entries_in_file = state.range(0);
+    auto file = get_file(n_entries_in_file);
+
+    for (auto _ : state)
+    {
+        std::istringstream istream{file};
+        // refilling streams introduces the same constant to
+        // both seqan3 and 2 benchmark runtime so they are still comparable
+
+        for (size_t i = 0; i < n_entries_in_file; ++i)
+            format.read(istream, options, seq, id, qual);
+
+        id.clear();
+        seq.clear();
+        qual.clear();
+    }
+}
+
+// ============================================================================
+// seqan2 comparison
+// ============================================================================
+#if SEQAN3_HAS_SEQAN2
+void fastq_read_from_stream_seqan2(benchmark::State & state)
+{
+    using namespace seqan;
+
+    std::istringstream istream{};
+    size_t n_entries_in_file = state.range(0);
+    auto file = get_file(n_entries_in_file);
+
+    auto restart_iterator = [&istream, &file]()    // c.f. format_fasta_benchmark
+    {
+        istream = std::istringstream{file};
+        // same constant as seqan3 benchmark
+
+        seqan::VirtualStream<char, Input> comp{};
+        open(comp, istream);
+        return directionIterator(comp, Input());
+    };
+
+    String<char> id{};
+    String<Dna5> seq{};
+    String<char> qual{};
+
+    for (auto _ : state)
+    {
+        auto it = restart_iterator();
+
+        for (size_t i = 0; i < n_entries_in_file; ++i)
+            readRecord(id, seq, qual, it, seqan::Fastq{});
+
+        clear(id);
+        clear(seq);
+        clear(qual);
+    }
+}
+
+#endif
+
+// ============================================================================
+// read dummy fastq file from temporary file on disk
+// ============================================================================
+void fastq_read_from_disk(benchmark::State & state)
+{
+    sequence_file_format_fastq format;
+
+    // create temporary file, automatically removed on destruction
+    test::tmp_filename file_name{"tmp.fastq"};
+    auto tmp_path = file_name.get_path();
+
+    std::ofstream ostream{tmp_path};
+    ostream << get_file(state.range(0));
+    ostream.close();
+
+    // benchmark
+    std::string id{};
+    std::vector<dna5> seq{};
+    std::vector<phred42> qual{};
```

Comment:
Hmm aren't these variables unused?

We should get a warnings because of unused variables. :thinking: